sdcm/utils/raft/__init__.py

```diff
@@ -0,0 +1,206 @@
+import contextlib
+import logging
+
+from typing import Protocol
+
+from sdcm.sct_events.database import DatabaseLogEvent
+from sdcm.sct_events.filters import EventsSeverityChangerFilter
+from sdcm.sct_events import Severity
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Group0MembersNotConsistentWithTokenRingMembersException(Exception):
+    pass
+
+
+class RaftFeatureOperations(Protocol):
+    _node: 'BaseNode'
+    enabled: bool = False
+
+    def get_status(self):
+        ...
+
+    def is_ready(self):
+        ...
+
+    def get_group0_members(self):
+        ...
+
+    def get_group0_non_voters(self):
+        ...
+
+    def clean_group0_garbage(self, raise_exception: bool = False):
+        ...
+
+    def diff_token_ring_group0_members(self) -> list[str]:
+        ...
+
+
+class Raft(RaftFeatureOperations):
+    def __init__(self, node: "BaseNode") -> None:
+        super().__init__()
+        self._node = node
+        self.enabled = True
```

Comment:
The `enabled` attribute value is expected to be immutable.
So, better to make it be a property:

```
@property
def is_enabled(self):
    return True
```

Note that it is better to use `is_` prefix for 2 reasons:
1) It shows state of the object returning `boolean` value.
2) To be consistent with other attributes of the same type -> `is_ready`.

Reply:
fixed